lib/Domain/Prototype/Type.php

```diff
@@ -14,14 +16,29 @@ final class Type extends Prototype
      */
     private $none = false;
 
-    public function __construct(string $type = null)
+    /**
+     * @var bool
+     */
+    private $nullable = false;
+
+    private function __construct(string $type = null, bool $nullable = false)
```

Comment:
It seems you're only using the named constructors so I thought we could set the constructor to private for a more secure code.

Reply:
This needs to be reverted I guess

Reply:
Oops, ignore that.